alphatwirl_interface/config/cutflow_stage.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import absolute_import
+import re
+import six
+import os
+import numpy as np
+from .base_stage import BaseStage
+from .config_exceptions import BadAlphaTwirlInterfaceConfig
+from ..selection import Selection
+
+
+__all__ = ["CutFlow"]
+
+
+class BadCutflowConfig(BadAlphaTwirlInterfaceConfig):
+    pass
+
+
+class BadSelectionFile(BadAlphaTwirlInterfaceConfig):
+    pass
+
+
+class MultipleWeightedSelectionsNotImplemented(UserWarning):
+    pass
+
+
+class CutFlow(BaseStage):
+    output_filename = "cut_flow.txt"
+
+    def apply_description(self, selection_file=None, selection=None, aliases=None,
+                          lambda_arg="ev", counter=True, counter_weights=None):
+        if not selection and not selection_file:
+            raise BadCutflowConfig("{}: Neither selection nor selection_file specified".format(self.name))
+        if selection and selection_file:
+            raise BadCutflowConfig("{}: Both selection and selection_file given. Choose one!".format(self.name))
+
+        if selection_file:
+            selection, aliases = _load_selection_file(self.name, selection_file)
+        self.selection = _prepare_selection(self.name, selection, lambda_arg, aliases)
+
+        self._counter = counter
+        if self._counter:
+            self._weights = _create_weights(self.name, counter_weights)
```

Comment:
Actually being able to use weighted cut flow counters will require use to merge #1 which needs a bit more work.  I'm tempted to let this merge in then sort #1 separately.